internal/gojs/logging/logging.go

```diff
@@ -90,42 +144,59 @@ func syscallValueCallResultLogger(ctx context.Context, mod api.Module, w logging
 	vRef := stack.ParamRef(0)               //nolint
 	m := stack.ParamString(mem, 1 /*, 2 */) //nolint
 
-	// TODO: add more than just filesystem
-	if vRef != goos.RefJsfs {
-		return
+	var resultNames []string
+	var resultVals []interface{}
+	switch vRef {
+	case goos.RefJsCrypto:
+		resultNames = custom.CryptoNameSection[m].ResultNames
+		rRef := stack.ParamVal(ctx, 6, gojs.LoadValue) // val is after padding
+		resultVals = []interface{}{rRef}
+	case goos.RefJsfs:
+		resultNames = custom.FsNameSection[m].ResultNames
+		resultVals = gojs.GetLastEventArgs(ctx)
+	default:
+		// TODO: other scopes
 	}
 
-	args := gojs.GetLastEventArgs(ctx)
-	argNames := custom.FsNameSection[m].ResultNames
-
 	w.WriteByte('(') //nolint
-	writeVals(w, args, argNames)
+	writeVals(w, resultNames, resultVals)
 	w.WriteByte(')') //nolint
 }
 
-func writeVals(w logging.Writer, vals []interface{}, names []string) {
+func writeVals(w logging.Writer, names []string, vals []interface{}) {
 	valLen := len(vals)
 	if valLen > 0 {
-		w.WriteString(names[0]) //nolint
-		w.WriteByte('=')        //nolint
-		// TODO: learn the types of the vals.
-		w.WriteString(fmt.Sprintf("%v", vals[0])) //nolint
+		writeVal(w, names[0], vals[0])
 		for i := 1; i < valLen; i++ {
-			switch names[i] {
+			name := names[i]
+			val := vals[i]
+
+			switch name {
 			case custom.NameCallback:
 				return // last val
 			case "buf": // always equal size with byteCount
 				continue
 			}
 
-			w.WriteByte(',')                          //nolint
-			w.WriteString(names[i])                   //nolint
-			w.WriteByte('=')                          //nolint
-			w.WriteString(fmt.Sprintf("%v", vals[i])) //nolint
+			w.WriteByte(',') //nolint
+			writeVal(w, name, val)
 		}
 	}
 }
 
+func writeVal(w logging.Writer, name string, val interface{}) {
+	if b, ok := val.(*goos.ByteArray); ok {
```

Comment:
exposed this (still internal) as there was no other way to access it in the logging package.